lib/charms/observability_libs/v1/kubernetes_service_patch.py

```diff
@@ -356,6 +357,36 @@ def _is_patched(self, client: Client) -> bool:
         ]  # noqa: E501
         return expected_ports == fetched_ports
 
+    def _on_upgrade_charm(self, event: UpgradeCharmEvent):
+        """Handle the upgrade charm event."""
+        # Handling the case when a user changes the service type from LB to ClusterIP in an upgrade, previous LB should be deleted.
+        if self.service_type == "ClusterIP":
+
+            client = Client()  # pyright: ignore
+
+            # Define a label selector to find services related to the app
+            selector: dict[str, Any] = {"app.kubernetes.io/name": self._app}
+
+            # Check if any service of type LoadBalancer exists
+            services = client.list(Service, namespace=self._namespace, labels=selector)
+            for service in services:
+                if (
+                    not service.metadata
+                    or not service.metadata.name
+                    or not service.spec
+                    or not service.spec.type
+                ):
+                    logger.warning("Skipping resource with incomplete metadata.")
```

Comment:
Do we need the warning? If yes, should we add more context?
```suggestion
                    logger.warning("Service patch: skipping resource with incomplete metadata: %s.", service)
```